src/System.IO.Compression/src/System/IO/Compression/DeflateZLib/Deflater.cs

```diff
@@ -172,12 +172,6 @@ internal bool Finish(byte[] outputBuffer, out int bytesRead)
         {
             Debug.Assert(null != outputBuffer, "Can't pass in a null output buffer!");
             Debug.Assert(outputBuffer.Length > 0, "Can't pass in an empty output buffer!");
-            Debug.Assert(NeedsInput(), "We have something left in previous input!");
```

Comment:
why are those asserts removed? they seem valid

Reply:
`public bool NeedsInput() => 0 == _zlibStream.AvailIn;` - this is incorrect though

Reply:
I removed these because we were hitting the debug statements in the case where the deflater is disposed with an ongoing WriteAsync. I added a test case for it.